In [1]:
import sys, os

# Go one level up from notebook to project root
project_root = os.path.abspath("..")
sys.path.append(project_root)

print("Project root added:", project_root)
print("Notebook CWD:", os.getcwd())

Project root added: c:\Users\AKALYA\OneDrive\Desktop\CODTECH-Internship-DataScience\Task4_Optimization
Notebook CWD: c:\Users\AKALYA\OneDrive\Desktop\CODTECH-Internship-DataScience\Task4_Optimization\notebook


In [2]:
import os

paths = [
    os.path.join(project_root, "data", "demand.csv"),
    os.path.join(project_root, "data", "capacity.csv"),
    os.path.join(project_root, "data", "costs.csv"),
]

for p in paths:
    print(p, "→", os.path.exists(p))

c:\Users\AKALYA\OneDrive\Desktop\CODTECH-Internship-DataScience\Task4_Optimization\data\demand.csv → True
c:\Users\AKALYA\OneDrive\Desktop\CODTECH-Internship-DataScience\Task4_Optimization\data\capacity.csv → True
c:\Users\AKALYA\OneDrive\Desktop\CODTECH-Internship-DataScience\Task4_Optimization\data\costs.csv → True


In [3]:
import pandas as pd

demand = pd.read_csv(paths[0])
capacity = pd.read_csv(paths[1])
costs = pd.read_csv(paths[2])

display(demand.head())
display(capacity.head())
display(costs.head())

product  demand
0       A     120
1       B      80
2       C     150
3       D     100
4       E      90

resource  available
0    labor_hours        600
1  machine_hours        400
2   raw_material        700
3         budget       1000

product  labor_hours_per_unit  machine_hours_per_unit  \
0       A                     2                       3   
1       B                     1                       2   
2       C                     4                       2   
3       D                     3                       4   
4       E                     5                       3   

   raw_material_per_unit  budget_per_unit  profit_per_unit  
0                      4                5               10  
1                      3                4                8  
2                      5                6               12  
3                      2                3                9  
4                      6                7               15

In [4]:
from src.model import build_and_solve

result = build_and_solve()
result

{'status': 'Optimal',
 'production_plan': {'A': 0.0, 'B': 10.0, 'C': 120.0, 'D': 35.0, 'E': 0.0},
 'total_profit': 1835.0}

In [18]:
print("Status:", result["status"])
print("Total Profit:", result["total_profit"])
print("Production Plan:")
for k, v in result["production_plan"].items():
    print(f"  {k}: {int(v) if v is not None else v}")

Status: Optimal
Total Profit: 1835.0
Production Plan:
  A: 0
  B: 10
  C: 120
  D: 35
  E: 0


In [19]:
import os
print(os.path.exists("reports/figures"))

True


In [20]:
from src.visualize import plot_production
plot_production(result["production_plan"])

In [21]:
os.listdir("reports/figures")

['production_plan.png']

## Executive Summary
- Solver Status: Optimal
- Maximum Profit: 1835.0
- Production Plan: A:0, B:10, C:120, D:35, E:0
- Insights: Product C dominates, B & D balance resources, A & E excluded.

![Production plan](notebook/reports/figures/production_plan.png)